ultrasonic/src/main/kotlin/org/moire/ultrasonic/view/SelectMusicFolderView.kt

```diff
@@ -0,0 +1,81 @@
+package org.moire.ultrasonic.view
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import android.widget.PopupMenu
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import org.moire.ultrasonic.R
+import org.moire.ultrasonic.domain.MusicFolder
+
+/**
+ * This little view shows the currently selected Folder (or catalog) on the music server.
+ * When clicked it will drop down a list of all available Folders and allow you to
+ * select one. The intended usage us to supply a filter to lists of artists, albums, etc
+ */
+class SelectMusicFolderView(
+    private val context: Context,
+    root: ViewGroup,
+    private val musicFolders: List<MusicFolder>,
+    private var selectedFolderId: String?,
+    private val onUpdate: (String, String?) -> Unit
+) : RecyclerView.ViewHolder(
+    LayoutInflater.from(context).inflate(
+        R.layout.select_folder_header, root, false
+    )
+) {
+    private val folderName: TextView = itemView.findViewById(R.id.select_folder_2)
+    private val layout: LinearLayout = itemView.findViewById(R.id.select_folder_header)
+
+    init {
+        if (selectedFolderId != null) {

```

Comment:
IMHO this needs an else case where the folderName is set to "All folders". Currently the text will be empty when the fragment is first started.